Typescript Migration - Backend

mikephu opened the pull request:
- Currently working on migrating controllers and api routing under packages/server to typescript.

lxkedinh committed:
feat: initial backend typescript setup
- prisma schema naming changes
- added typescript eslint parser

lxkedinh committed:
chore: Merge branch 'main' into backend-typescript-migration

lxkedinh committed:
chore: update yarn.lock

lxkedinh committed:
feat: created Request and Response types
- adjusted some typescript and eslint config properties
- adjust Prisma data model names

lxkedinh committed:
chore: changed backend eslint rules
- added yarn build command for building typescript into javascript

lxkedinh committed:
fix: typed express response type

mikephu committed:
migrated events/guilds controllers to typescript

lxkedinh commented on packages/server/controllers/guilds.ts:
```diff
@@ -0,0 +1,21 @@
+import prisma from "../utils/prisma";
+import { Guild } from ".prisma/client/index";
+
+async function getAllGuilds() {
+  const query = await prisma.guild.findMany();
+  return query;
+}
```
We should write down the return type annotations for all the controller functions for clarity.

lxkedinh commented on packages/server/controllers/events.ts:
```diff
@@ -1,16 +1,17 @@
 // const postgres = require("../utils/postgres");
-const prisma = require("../utils/prisma");
+import prisma from "../utils/prisma";
+import { Event } from ".prisma/client/index";
```
your imports for the `Event` type and `Guild` type in the `guilds.ts` file seem to be broken. They showed up as errors on my editor and the correct import was
```javascript
import { Event } from "@prisma/client";
import { Guild } from "@prisma/client";
```

These types are automatically generated by the Prisma client so make sure you have Prisma client generated by running `yarn workspace server prisma generate` in icebreak root directory.

lxkedinh reviewed (changes requested):
Really good job so far Michael, there are only a few minor things that need to be changed.

mikephu committed:
fixed imports and function typing

lxkedinh committed:
feat: migrated users controller

lxkedinh committed:
chore: Merge branch 'main' into backend-typescript-migration

lxkedinh committed:
fix: fail response body type

lxkedinh committed:
fix: adjusted export

mikephu committed:
fix: adjust exports in events/guild controllers

mikephu committed:
feat: migrate events/guilds routes

lxkedinh committed:
feat: migrated auth controller and jwt utility functions
- added ts-node to run typescript directly without having to transpile to js first

lxkedinh committed:
chore: Merge branch 'main' into backend-typescript-migration

lxkedinh committed:
feat: migrated index.js and users api router
- fixed ApiRequest type to default to empty object instead of void to fix type error

mikephu committed:
fix: fixed ESlint errors

lxkedinh committed:
fix: migrated local auth api route
- enables API testing since local auth is used in automatic authentication in our Postman
- fixed Guild Prisma model not mapping correctly to guilds table in Postgres

lxkedinh committed:
feat: migrated auth api routes

lxkedinh committed:
chore: updated with main
